tests/utils.py

```diff
@@ -27,13 +28,17 @@ def run(
         **kwargs
     ):
 
-    with benchmarks:
+    callbacks = kwargs.get('callbacks', [])
+    history = HistoryCallback()
+    callbacks.append(history)
+
+    with benchmarks(history):
```

Comment:
To merge the changes from `main` I had to slightly awkwardly append the history callback to any other callbacks already provided.